flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/TableEnvironmentTest.java

```diff
@@ -78,6 +78,19 @@ public void testConnect() {
         assertCatalogTable(CatalogTableImpl.fromProperties(table.toProperties()));
     }
 
+    @Test
+    public void testTableFromDescriptor() {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+
+        final Schema schema = Schema.newBuilder().column("f0", DataTypes.INT()).build();
+        final TableDescriptor descriptor =
+                TableDescriptor.forConnector("fake").schema(schema).build();
+
+        Table table = tEnv.from(descriptor);
+        assertEquals(
```

Comment:
this tests only the schema, isn't it possible to access the underlying catalog table?

Reply:
Do you mean `getQueryOperation`? That'll be a `CatalogQueryOperation` which only has the identifier and resolved schema. I don't think there's a way to the underlying table (other than looking it up ourselves), but even so we have no table to compare it against; we'd need to assert individual parts anyway, no?

Reply:
I thought we have access to the original `CatalogTable` in the stack. But the identifier should be enough to look it up in the catalog manager and verify that `connector=fake` has ended up in the right location.